apps/oxlint/src/walk.rs

```diff
@@ -174,6 +173,53 @@ mod test {
     }
 
     #[test]
+    fn test_parent_gitignore_doesnt_break_subdirectory_walk() {
+        // This test verifies the fix for a v1.36.0 regression where oxlint found 0 files
+        // when walking subdirectories.
+        // The bug was caused by .require_git(false), which caused the walker to traverse
+        // upward and find parent .gitignore files. When oxc's root .gitignore has
+        // "/*.js", those patterns incorrectly filtered out files in subdirectories,
+        // resulting in 0 files found.
+        // This test must run from within the oxc repo to reproduce the bug, since it relies
+        // on finding the real .gitignore in the parent directory structure.
```

Comment:
Hmm, do we have a way to reproduce this bug that doesn't rely on the gitignore at the root of the repo? It'd be goofy to change that and break tests.

Reply:
That's a fair point. I'll see if I can push up something to address this shortly.